addons/metadata.demo.movies/demo.py

```diff
@@ -1,98 +1,143 @@
-        liz.setAvailableFanart([{'image': 'DefaultBackFanart.png', 'preview': 'DefaultBackFanart.png'}, 
-                                {'image': '/home/akva/Pictures/hawaii-shirt.png', 'preview': '/home/akva/Pictures/hawaii-shirt.png'}])
-        xbmcplugin.setResolvedUrl(handle=int(sys.argv[1]), succeeded=True, listitem=liz)
+        list_item.setRating("imdb", 9, 100000, True )
+        list_item.setRating("themoviedb", 8.9, 1000)
+        list_item.setUniqueIDs({ 'imdb': 'tt8938399', 'tmdb' : '9837493'}, 'imdb')
+        list_item.setCast(
+            [
+                {
+                    'name': 'spiff', 'role': 'himself', 
+                    'thumbnail': '/home/akva/Pictures/fish.jpg', 'order': 2
+                }, {
+                    'name': 'monkey', 'role': 'orange', 
+                    'thumbnail': '/home/akva/Pictures/coffee.jpg', 'order': 1
+                }
+            ]
+        )
+        list_item.addAvailableArtwork('DefaultBackFanart.png', 'banner')
+        list_item.addAvailableArtwork(
+            '/home/akva/Pictures/hawaii-shirt.png', 'poster'
+        )
+        list_item.setAvailableFanart(
+            [
+                {
+                    'image': 'DefaultBackFanart.png', 
+                    'preview': 'DefaultBackFanart.png'
+                }, {
+                    'image': '/home/akva/Pictures/hawaii-shirt.png', 
+                    'preview': '/home/akva/Pictures/hawaii-shirt.png'
+                }
+            ]
+        )
+        setResolvedUrl(handle = int(argv[1]), succeeded = True, listitem = list_item)
 elif action == 'getartwork':
-    url=urllib.unquote_plus(params["id"])
+    url = unquote_plus(params["id"])
     if url == '456':
-        liz=xbmcgui.ListItem('Demo movie 1', offscreen=True)
-        liz.addAvailableArtwork('DefaultBackFanart.png', 'banner')
-        liz.addAvailableArtwork('/home/akva/Pictures/hawaii-shirt.png', 'poster')
-        liz.setAvailableFanart([{'image': 'DefaultBackFanart.png', 'preview': 'DefaultBackFanart.png'}, 
-                                {'image': '/home/akva/Pictures/hawaii-shirt.png', 'preview': '/home/akva/Pictures/hawaii-shirt.png'}])
-        xbmcplugin.setResolvedUrl(handle=int(sys.argv[1]), succeeded=True, listitem=liz)
+        list_item = ListItem('Demo movie 1', offscreen = True)
+        list_item.addAvailableArtwork('DefaultBackFanart.png', 'banner')
+        list_item.addAvailableArtwork('/home/akva/Pictures/shirt.png', 'poster')
+        list_item.setAvailableFanart(
+            [
+                {
+                    'image': 'DefaultBackFanart.png', 
+                    'preview': 'DefaultBackFanart.png'
+                }, {
+                    'image': '/home/akva/Pictures/hawaii-shirt.png', 
+                    'preview': '/home/akva/Pictures/hawaii-shirt.png'
+                }
+            ]
+        )
+        setResolvedUrl(handle = int(argv[1]), succeeded = True, listitem = list_item)
```

Comment:
I thikn we should also set the "path" for an item in order ot use `setResolvedUrl()`?